### PR TITLE
Add cloud_init argument to image creation

### PR DESCRIPTION
**Pull request by stvnjacobs**
## 📝 Description

This adds the `cloud_init` flag to the image create step.

## ✔️ How to Test

There are no unit or integration tests included with this patch. I am putting it up for consideration before those are completed.

To test manually, and to get a working cloud-init image, an image can be created using an existing image which has the cloud-init capability.

```console
$ linode-cli images list --text --is_public true --format id,capabilities | awk '/cloud-init/ { print $1 }'
linode/debian11
linode/ubuntu20.04
linode/ubuntu22.04
linode/ubuntu23.10
```

## 📷 Preview

```hcl
locals { timestamp = regex_replace(timestamp(), "[- TZ:]", "") }

source "linode" "demo" {
  image          = "linode/debian11"
  image_label    = "demo-debian11"
  instance_label = "pkr-demo-debian11-${local.timestamp}"
  instance_type  = "g6-nanode-1"
  region         = "us-iad"
  ssh_username   = "root"
  cloud_init     = true
}
```

### Config structure

I was uncertain on the best way to format the config. I am sure I am over thinking it, but what good does that do if I don't share where my head is at.

[`POST /v4/images`](https://www.linode.com/docs/api/images/#image-create): When an image is created, `cloud-init` is a boolean.

[`GET /v4/images/{imageId}`](https://www.linode.com/docs/api/images/#image-view): When an image is viewed, `cloud-init` is one of an array of `capabilities`.

I was unsure whether to structure the config so it matches the create arguments:

```hcl
source "linode" "demo" {
  image      = "linode/debian11"
  # ...
  cloud_init = true
}
```

Or to model it like the image resource it creates. 

```hcl
source "linode" "demo" {
  image        = "linode/debian11"
  # ...
  capabilities = ["cloud-init"]
}
```

In these situations I typically refer to official hashicorp cloud provider plugins, but I couldn't really find anything comparable.

### Unfamiliar with codebase

I admit to being a bit rusty around these parts. the `hcl2spec.go` was new to me, and in `config.go` is wasn't sure if `required:"false"` was really needed. I am just committing this as a proof of what worked for me, not what is fully correct.

### CODE REVIEW

**Comment by lgarber-akamai** [comment]
@stvnjacobs I think the first structure you mentioned (`cloud_init = true`) is ideal since Packer isn't stateful 👍

**Review by lgarber-akamai — APPROVED** [review]
LGTM! This PR doesn't add proper E2E testing for `cloud_init` given #185 hasn't been merged quite yet, but the change is simple enough that unit tests should be sufficient 👍

File: .web-docs/components/builder/linode/README.md [hunk 1]
```diff
@@ -97,6 +97,8 @@ can also be supplied to override the typical auto-generated key:
   Please note that when you create a new Linode instance with a private image, you will
   be required to setup a new root password.
 
+- `cloud_init` (string) - Whether the newly created image supports cloud-init.
```
Comment (zliang-akamai) [inline .web-docs/components/builder/linode/README.md:100]:
```suggestion
- `cloud_init` (bool) - Whether the newly created image supports cloud-init.
```

File: docs/builders/linode.mdx [hunk 2]
```diff
@@ -103,6 +103,8 @@ can also be supplied to override the typical auto-generated key:
   Please note that when you create a new Linode instance with a private image, you will
   be required to setup a new root password.
 
+- `cloud_init` (string) - Whether the newly created image supports cloud-init.
```
Comment (zliang-akamai) [inline docs/builders/linode.mdx:106]:
```suggestion
- `cloud_init` (bool) - Whether the newly created image supports cloud-init.
```

Comment (lgarber-akamai) [reply docs/builders/linode.mdx:106]: Oops, thanks for catching this!

**Comment by lgarber-akamai** [comment]
Failing tests are unrelated to this PR 🙂

**Review by ykim-1 — APPROVED** [review]

**Review by zliang-akamai — APPROVED** [review]

**Pull request merged**